arrays_lab.js

```diff
@@ -0,0 +1,256 @@
+
+let arr3 = ["strawberry", "banana"];
+console.log(arr3);
+arr3.push("mango");
+console.log(arr3);
+
+// 4. Add "blueberry" to the front of the same fruits array.
+
+arr3.unshift("blueberry");
+console.log(arr3);
+
+// 5. Remove the last element of the fruits array.
+
+arr3.pop();
+console.log(arr3);
+
+// 6. console.log the length of the fruits array
+
+console.log(arr3.length);
+
+// 7. Remove the first element of the fruits array.
+
+arr3.shift();
+console.log(arr3);
+
+// 8. Join all the elements in the fruits array with a '$'.
+
+console.log(`${arr3}`);
+
+// 9. Given two arrays, write code that logs the larger array.
+// If the arrays are the same length, log "They are the same size"
+//
+// ```js
+// let firstArr = [1,2,3]
+// let secondArr = [1,5,2,4]
+//
+// // log [1,5,2,4]
+// ```
+
+let firstArr = [45, 53, 67, 23, 7, 928];
+let secondArr = [3, 13, 82, 5]
+
+if (firstArr.length > secondArr.length) {
+  console.log(firstArr);
+} else if (firstArr.length < secondArr.length) {
+  console.log(secondArr);
+} else {
+  console.log("They are the same.");
+}
+
+// 10. Given an array, log its middle element.  If the middle of the
+// array is between two elements, log both of them.
+//
+// ```js
+// let hasExactMiddleArr = [1,5,3]
+//
+// // log 5
+//
+// let hasNoExactMiddleArr = [1,4,6,9]
+//
+// // log 4,6
+//
+// ```
+
+let middleArr = [7, 6, 3, 0, 2];
+let noMiddleArr = [2, 5, 1, 7, 4, 8];
+
+let middleIndex = middleArr[Math.floor(middleArr.length / 2)]
+// let middleIndex2 =
+console.log(middleIndex);
+console.log(noMiddleArr[(noMiddleArr.length / 2) - 1] + ", " +
+ noMiddleArr[(noMiddleArr.length / 2)]);
+
+
+// console.log(noMiddleArr[Math.floor(noMiddleArr.length / 2)],
+// noMiddleArr[Math.ceil(noMiddleArr.length / 2)]);
+
+// 11. Write code that logs every value in an array using a for loop
+//
+// ```js
+// let logMeOutFor = ["I", "am", "a", "happy", "array"]
+//
+// /* I
+//   am
+//   a
+//   happy
+//   array
+// */
+// ```
+
+let arr4 = ["spaghetti", "fettucine", "farfalle", "lasanga", "penne"]
+for (let i = 0; i < arr4.length; i++){
+  console.log(arr4[i])
+}
+
+// 12. Write code that logs every value in an array using a while loop
+//
+// ```js
+// let logMeOutWhile = ["I", "am", "a", "happy", "array"]
+//
+// /* I
+//   am
+//   a
+//   happy
+//   array
+// */
+// ```
+let i = 0;
+while (i < arr4.length) {
+  console.log(arr4[i]);
+  i++
+}
+
+//
+// 13. Write code that logs every value in an array using a for/of loop
+//
+// ```js
+// let logMeOutForOf = ["I", "am", "a", "happy", "array"]
+//
+// /* I
+//   am
+//   a
+//   happy
+//   array
+// */
+// ```
+for (let x of arr4) {
+  console.log(x);
+}
+//
+// 14. Given an array of numbers, write code that logs the sum of all
+// the numbers
+//
+// ```js
+// let firstArrToSum = [1, 2, 3, 4, 5, 6]
+// //  21
+//
+// let secondArrToSum = [1, 2, 3, 4, 5, -6]
+// //  9
+// ```
+
+let firstArrToSum = [32, 3, 45, 7]
```

Comment:
What would happen if we change array length to 7? Would this code still give us the same result? This is practice called hard coding which is not a good practice. Your code must be able to accommodate for differing inputs.